dotcom-rendering/.storybook/decorators/themeDecorator.tsx

```diff
@@ -27,7 +27,7 @@ const lightStoryCss = css`
  */
 const colourSchemeDecorator =
 	(colourScheme: 'light' | 'dark') =>
-	(formats: ArticleFormat[]): Decorator =>
+	<Args = StrictArgs,>(formats: ArticleFormat[]): Decorator<Args> =>
```

Comment:
What does this do? Can’t see [any info in the docs](https://storybook.js.org/docs/writing-stories/typescript#typing-custom-args).

Reply:
Yeah I had trouble finding information in the docs too, so I looked at the type definitions. I wanted a way to enforce type safety for the `context` parameter in the `HydratedLayout` decorator, because I needed to pull out specific values from the args, like `article`, and set specific values for the args, like `NAV`.

The `Decorator` type appears to take a parameter for the args of the `Story`, so if you parameterise it with the prop types of the component you're working with, you get type safety when working with the `context` parameter.

Given that `HydratedLayout` also derives the `format` for these stories, I wanted to bundle usage of the `colourSchemeDecorator` in, so that consumers don't have to manually include at the story level. That avoids having to specifying format in two places for each story, which would cause bugs if one didn't match the other.

To do this, I needed a way to specify the type parameter for the `colourSchemeDecorator` too, otherwise you get a type mismatch with `StrictArgs`, which is the default parameter when unspecified.